lib/barbeque_client/executor.rb

```diff
@@ -8,9 +8,19 @@ class Executor
     # @param [String] queue_name - barbeque's job_queues.name
     def initialize(job:, message:, message_id:, queue_name:)
       @job        = job
-      @message    = JSON.load(message)
       @message_id = message_id
       @queue_name = queue_name
+
+      parsed_message = JSON.load(message)
+
+      # `arguments` in ActiveJob::Base.execute is expected as Array
+      # and it expands to the arguments for AJ::Base#perform.
+      # So when message is not an Array, it's converted to a 1-element Array.
```

Comment:
In this comment, since ActiveJob will always send message in an Array, why such message comes sounds ambiguous.